src/test/datascience/data-viewing/dataViewerDependencyService.unit.test.ts

```diff
@@ -4,118 +4,136 @@
 'use strict';
 
 import { assert } from 'chai';
-import * as path from '../../../platform/vscode-path/path';
-import { SemVer } from 'semver';
-import { anything, deepEqual, instance, mock, verify, when } from 'ts-mockito';
+import { anything, instance, mock, when } from 'ts-mockito';
 import { ApplicationShell } from '../../../platform/common/application/applicationShell';
 import { IApplicationShell } from '../../../platform/common/application/types';
-import { PythonExecutionFactory } from '../../../platform/common/process/pythonExecutionFactory.node';
-import { IPythonExecutionFactory, IPythonExecutionService } from '../../../platform/common/process/types.node';
+import {
+    DataViewerDependencyService,
+    kernelGetPandasVersion
+} from '../../../webviews/extension-side/dataviewer/dataViewerDependencyService';
+import { IKernel } from '../../../kernels/types';
 import { Common, DataScience } from '../../../platform/common/utils/localize';
-import { IInterpreterService } from '../../../platform/interpreter/contracts';
-import { PythonEnvironment } from '../../../platform/pythonEnvironments/info';
-import { ProductInstaller } from '../../../kernels/installer/productInstaller.node';
-import { IInstaller, Product } from '../../../kernels/installer/types';
-import { DataViewerDependencyService } from '../../../webviews/extension-side/dataviewer/dataViewerDependencyService.node';
-import { Uri } from 'vscode';
+import * as helpers from '../../../kernels/helpers';
+import * as sinon from 'sinon';
 
 suite('DataScience - DataViewerDependencyService', () => {
     let dependencyService: DataViewerDependencyService;
     let appShell: IApplicationShell;
-    let pythonExecFactory: IPythonExecutionFactory;
-    let installer: IInstaller;
-    let interpreter: PythonEnvironment;
-    let interpreterService: IInterpreterService;
-    let pythonExecService: IPythonExecutionService;
+    let kernel: IKernel;
+
     setup(async () => {
-        interpreter = {
-            displayName: '',
-            uri: Uri.file(path.join('users', 'python', 'bin', 'python.exe')),
-            sysPrefix: '',
-            sysVersion: '',
-            version: new SemVer('3.3.3')
-        };
-        pythonExecService = mock<IPythonExecutionService>();
-        installer = mock(ProductInstaller);
         appShell = mock(ApplicationShell);
-        pythonExecFactory = mock(PythonExecutionFactory);
-        interpreterService = mock<IInterpreterService>();
-
-        dependencyService = new DataViewerDependencyService(
-            instance(appShell),
-            instance(installer),
-            instance(pythonExecFactory),
-            instance(interpreterService),
-            false
-        );
+        kernel = instance(mock<IKernel>());
+        dependencyService = new DataViewerDependencyService(instance(appShell), false, undefined);
```

Comment:
These are the kernel-specific tests. This code works in Node and in the browsers.